src/modules/map/components/openlayers/OpenLayersMap.vue

```diff
@@ -172,8 +180,30 @@ export default {
             }
             return null
         },
+        isBackgroundVectorTile() {
+            return (
+                this.currentBackgroundLayer &&
+                this.currentBackgroundLayer.type === LayerTypes.VECTOR
+            )
+        },
+        backgroundLayerOnTopOfVectorBackground() {
+            // we currently only have the case of a light base map vector tile background with our national map on top
+            if (
+                this.isBackgroundVectorTile &&
+                this.currentBackgroundLayer.getID() === VECTOR_TILES_STYLE_ID
+            ) {
+                return this.backgroundLayers.find(
+                    (layer) => layer.getID() === 'ch.swisstopo.pixelkarte-farbe'
+                )
+            }
+            return null
+        },
         // zIndex calculation conundrum...
         startingZIndexForVisibleLayers() {
+            // checking if the BG layer is a vector layer that has anoter
+            if (this.isBackgroundVectorTile && this.backgroundLayerOnTopOfVectorBackground) {
+                return 2
+            }
             return this.currentBackgroundLayer ? 1 : 0
         },
```

Comment:
I think we could do this a bit more generic. If I understood correctly we already have in the config a flag telling if the layer is a background or not ? We could extend this flag to an int specifying the bg priority/zindex; e.g. 
- `background = 0` => not a background
- `background = 1` => upper background
- `background = 2` => lower background 
- ...

Reply:
I can't just set fixed value and not use the index, otherwise OL complains that it's not possible to insert a layer at an index greater than the current length of the array... Hence this "complicated" zIndex calculation

Reply:
So I have to dynamically have a sequence of index, where all are used to insert a layer, starting from the small value to the biggest

Reply:
👍🏼